internals/daemon/daemon.go

```diff
@@ -368,6 +380,27 @@ func (d *Daemon) Init() error {
 		logger.Noticef("HTTP API server listening on %q.", d.httpAddress)
 	}
 
+	if d.httpsAddress != "" {
+		tlsConf := &tls.Config{
+			// Enable server support for HTTP1.1 and HTTP2 over TLS. The server
+			// will only pick HTTP2 if the client indicated the same support in
+			// the client hello message. The Pebble client will not pick HTTP2
+			// because it requires integrated websockets to work, which does
+			// not support HTTP2 (it does not yet implement RFC8441). However,
+			// external clients, such as curl, will be able to switch to HTTP2
+			// for non-websocket dependant API calls.
+			NextProtos:     []string{"h2", "http/1.1"},
+			MinVersion:     tls.VersionTLS13,
+			GetCertificate: d.overlord.TLSManager().GetCertificate,
+		}
```

Comment:
Can the TLSManager have a `ListenConfig() tls.Config` method instead? Or something similar?

Reply:
Added this in the TLS Manager PR. Great idea! Resolving.